pkg/sdkserver/sdkserver.go

```diff
@@ -246,17 +247,17 @@ func (s *SDKServer) Run(ctx context.Context) error {
 		return err
 	}
 
-	logLevel := agonesv1.SdkServerLogLevelInfo
-	// grab configuration details
-	if gs.Spec.SdkServer.LogLevel != "" {
-		logLevel = gs.Spec.SdkServer.LogLevel
+	var logLevel string
```

Comment:
So the one that really needs changing is here:

https://github.com/googleforgames/agones/blob/8b2c44256954cd449e6a39ad68f1da767c80e927/cmd/sdk-server/main.go#L72-L75

Which is some of the Info level output that was happening when logging was set to Warning/Error only.

What I'm thinking we should do here, is take a `logrus.Level` as part of `NewSDKServer` (which will mean refactoring some unit tests, and some error handling code) and passing it in after parsing in main.go.

Then we can set the `s.logger.Logger.SetLevel(level)` right under:
https://github.com/googleforgames/agones/pull/3631/files#diff-2b47120b02bc51219ccd6d97b8fe68f9d5043997eba480c0655babcafd771e4eR185 (line 185), so that is also covers all logging in this file.

Hopefully that all was clear!

Reply:
Following your guidance, I made below modifications:
1. Removed the hardcoded log level in `controller.go` and replaced it with `gs.Spec.SdkServer.LogLevel`.
2. Added log level as a parameter in the `NewSDKServer` function to accept the log level.
3. Added log level parsing in `main.go` and passed it to `NewSDKServer` function.
4. Set `s.logger.Logger.SetLevel(logLevel)` right after its initialization  in the `NewSDKServer` function.
5. Modified the `controller_test.go` and `sdkserver_test.go` files based on the lint suggestion.

Please let me know if anything missing